lib/src/models/scoped_class_declaration.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:meta/meta.dart';
+
+import 'class_type.dart';
+
+/// Represents a declaration of a class / mixin / extension.
+@immutable
+class ScopedClassDeclaration {
+  /// Type of declared class entity.
+  final ClassType type;
+
+  /// The node that represents in the AST structure for a Dart program.
+  final CompilationUnitMember declaration;
+
+  /// Returns user defined name.
+  String get name {
+    if (declaration is ExtensionDeclaration) {
```

Comment:
Seems that if you introduce a local variable like `final declaration = this.declaration;` and then type check it, the analyzer will not require a type cast.